### ISSUE TITLE
BulkDescriptor.DeleteMany requires <T> while not required for overload IEnumerable<string> ids

### ISSUE THREAD

**Issue by octocat-mona** [elastic/elasticsearch-net#3500]
**NEST/Elasticsearch.Net version**:
6.3.1

**Elasticsearch version**:
6.5.0

**Description of the problem including expected versus actual behavior**:
There are multiple overloads for `BulkDescriptor.DeleteMany` which can be used in for example the `BulkAsync` method:
- `DeleteMany<T>(IEnumerable<T> @objects, Func<BulkDeleteDescriptor<T>, T, IBulkDeleteOperation<T>> bulkDeleteSelector = null)` ([source](https://github.com/elastic/elasticsearch-net/blob/master/src/Nest/Document/Multiple/Bulk/BulkRequest.cs#L61))
- `DeleteMany<T>(IEnumerable<string> ids, Func<BulkDeleteDescriptor<T>, string, IBulkDeleteOperation<T>> bulkDeleteSelector = null)` ([source](https://github.com/elastic/elasticsearch-net/blob/master/src/Nest/Document/Multiple/Bulk/BulkRequest.cs#L71))

Note that both have a `bulkDeleteSelector` parameter which is optional.
Now I want to use the 2nd `IEnumerable<string> ids` overload without the optional `bulkDeleteSelector` parameter, for example:

```csharp
IEnumerable<string> ids = new List<string> { "1", "2" };
var bulkResponse = await client.BulkAsync(descriptor => descriptor
    .Index("my-index")
    .DeleteMany(ids)
);
```
But because I didn't provide a `<T>` C# selects the `IEnumerable<T> @objects` overload instead, causing the operation to fail.
Because I don't need the optional `bulkDeleteSelector` parameter (and it's not returned either) I can workaround this issue by using any `<T>` like this:

```csharp
var bulkResponse = await client.BulkAsync(descriptor => descriptor
    .Index("my-index")
    .DeleteMany<object>(ids)
);
```

**Steps to reproduce**:
 1. Call `BulkAsync` with `IEnumerable<string> ids` as input, but without a `<T>`
 2. Bulk call to ES fails

I think the solution could be quite simple: add an extra overload without the `bulkDeleteSelector` overload and remove the default `= null` for the original overload.

**Comment by Mpdreamz** [comment]
Thanks for raising this @kpstolk I can see how this is problematic. I am not sure we can fix this in 6.x without breaking binary backwards compatibility but we'll try and fix this for the next major release.

**Comment by kpstolk** [comment]
@Mpdreamz Although it's easy to workaround this issue but would be nice indeed to have this fixed 👍

**Comment by stevejgordon** [comment]
I've addressed the API design for the v8.0.0 with some further changes which will appear in the next alpha. I'm going to close this on that basis.
